File: Python/Arrays/ReverseArray.py
# Time -Complexity-O(n)
# Space Complexity-O(1)
# Two Pointer Approach

class Solution:
    def reverseArray(self, arr):
        n=len(arr)
        i=0
        j=n-1
        while i <= j:
            arr[i],arr[j]=arr[j],arr[i]
            i=i+1
            j=j-1
        return arr
            
        
        
        

File: C++/Arrays/ReverseArray.cpp
//optimal(2 pointer)
//Time Complexity=O(n)
//Space Complexity=O(1)
class Solution {
  public:
    void reverseArray(vector<int> &arr) {
        int n=arr.size();
        int l=0;
        int r=n-1;
        while(l<r){
            swap(arr[l],arr[r]);
            l++;
            r--;
        }
    }
};

//stl approach
//Time Complexity=O(n)
//Space Complexity=O(1)
class Solution {
  public:
    void reverseArray(vector<int> &arr) {
      
        reverse(arr.begin(),arr.end());
    }
};